### PR TITLE
Some minor cleanup post the addition of TYP_SIMD64 and ZMM support - P1

### PR DESCRIPTION
**Pull request by tannergooding**
The initial support was added in https://github.com/dotnet/runtime/pull/80960 and a few unrelated cleanup opportunities were seen.

This is the first cleanup and covers the simpler things which can be easily relegated to TARGET_XARCH only.

There will be a separate PR cleaning up the SIMD handle logic to reduce code duplication and then another which ensures that `simd32_t` and `simd64_t` are xarch only as well.

### CODE REVIEW

**Comment by tannergooding** [comment]
Minor TP impact is because we need to call `compVerifyInstructionSetUnusable(InstructionSet_AVX512F);` as part of `maxSIMDStructBytes`.

This is called for each `SimdOrHWSimdClass` field as part of `Compiler::StructPromotionHelper::GetFieldInfo` in `lclvars.cpp`

We only need to report that info "once" per compilation, so we should probably look at refactoring the `maxSIMDStructBytes` to record this once -or- ensure that callers (such as `StructPromotionHelper`) cache the info they need so it doesn't need to be repeatedly done...

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -9184,14 +9151,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         {
             return true;
         }
-        if (JitConfig.JitStressEvexEncoding() && compOpportunisticallyDependsOn(InstructionSet_AVX512F_VL))
```
Comment (kunalspathak) [inline src/coreclr/jit/compiler.h:9187]: Why is this removed?

File: src/coreclr/jit/codegenxarch.cpp [hunk 2]
```diff
@@ -10709,7 +10709,6 @@ void CodeGen::genZeroInitFrameUsingBlockInit(int untrLclHi, int untrLclLo, regNu
     assert(compiler->compGeneratingProlog);
     assert(genUseBlockInit);
     assert(untrLclHi > untrLclLo);
-    assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
```
Comment (kunalspathak) [inline src/coreclr/jit/codegenxarch.cpp:10712]: Replacing this assert is not needed? Is the idea that if AVX2 is not supported, then the default is SSE2 now?

**Review by kunalspathak — APPROVED** [review]
LGTM. Added couple of questions for better understanding.

File: src/coreclr/jit/compiler.h (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/compiler.h:9187]: Shouldn't have been, was a refactoring mistake

Comment (tannergooding) [reply src/coreclr/jit/compiler.h:9187]: Fixed.

File: src/coreclr/jit/codegenxarch.cpp (hunk 2 above)
Comment (tannergooding) [reply src/coreclr/jit/codegenxarch.cpp:10712]: The default has been SSE2 for many years (always for x64 and since around .NET Core 2.1 for x86). We require it and consider it part of the baseline ISA, so there is no need to check for its existence outside of supporting the `DOTNET_EnableSSE2=0` switch for HWIntrinsic importation.
